invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/AccessField/components/MetadataAccess.js

```diff
@@ -12,15 +12,41 @@ import { i18next } from "@translations/invenio_rdm_records/i18next";
 import Overridable from "react-overridable";
 
 export const MetadataAccess = (props) => {
-  const { recordAccess, communityAccess } = props;
+  const {
+    recordAccess,
+    communityAccess,
+    record,
+    recordRestrictionGracePeriod,
+    allowRecordRestriction,
+  } = props;
   const publicMetadata = recordAccess === "public";
   const publicCommunity = communityAccess === "public";
 
+  const checkGracePeriod = () => {
+    const created = record?.created;
+    if (!allowRecordRestriction && created) {
+      const createdDate = new Date(created);
+
+      // calculate end of grace period
+      createdDate.setDate(createdDate.getDate() + recordRestrictionGracePeriod);
+
+      return createdDate.getTime() <= new Date().getTime();
+    }
+
+    return false;
+  };
+
+  const isButtonDisabled = () => {
+    return checkGracePeriod() && record?.access?.record !== "restricted";
```

Comment:
this will avoid calling the grace period function when is it not needed. I moved those conditions from checkGracePeriod
```suggestion
  const isButtonDisabled = () => {
    if (!allowRecordRestriction && record && publicMetadata){
        return checkGracePeriod();
    }
```